In [1]:
import numpy as np
import pandas as pd 
import warnings
warnings.filterwarnings('ignore')

<h1> UNITED KINGDOM ROAD ACCIDENT </h1>
<hr>
<h2> Maranan Reilh Rich P. </h2>
<h2> BSIT-3A </h2>

In [2]:
uk_accident = pd.read_csv('ds\\uk_road_accident.csv')

In [3]:
uk_accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [4]:
uk_accident.columns

Index(['Index', 'Accident_Severity', 'Accident Date', 'Latitude',
       'Light_Conditions', 'District Area', 'Longitude',
       'Number_of_Casualties', 'Number_of_Vehicles', 'Road_Surface_Conditions',
       'Road_Type', 'Urban_or_Rural_Area', 'Weather_Conditions',
       'Vehicle_Type'],
      dtype='object')

In [5]:
uk_accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [6]:
uk_accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [7]:
uk_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [8]:
print(uk_accident.columns)

Index(['Index', 'Accident_Severity', 'Accident Date', 'Latitude',
       'Light_Conditions', 'District Area', 'Longitude',
       'Number_of_Casualties', 'Number_of_Vehicles', 'Road_Surface_Conditions',
       'Road_Type', 'Urban_or_Rural_Area', 'Weather_Conditions',
       'Vehicle_Type'],
      dtype='object')


In [9]:
uk_accident ['Accident_Severity']

0         Serious
1         Serious
2         Serious
3         Serious
4          Slight
           ...   
660674     Slight
660675     Slight
660676     Slight
660677    Serious
660678    Serious
Name: Accident_Severity, Length: 660679, dtype: object

In [10]:
uk_accident ['Accident_Severity'] == 'Serious'

0          True
1          True
2          True
3          True
4         False
          ...  
660674    False
660675    False
660676    False
660677     True
660678     True
Name: Accident_Severity, Length: 660679, dtype: bool

In [11]:
uk_accident[uk_accident['Accident_Severity'] == 'Serious']

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
5       200701BS67159           Serious    18-09-2019  51.497750   
...               ...               ...           ...        ...   
660629  201091NL00686           Serious     11/2/2022  57.687933   
660645  201091NL05842           Serious    18-10-2022  60.162660   
660672  201091NM01085           Serious    29-01-2022  57.170725   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
5                          Daylight  Kensington and Chelsea  -0.192556   
...                             ...                     ...        ...   
660629                     Daylight        Shetland Islands  -4.176860   
660645                     Daylight        Shetland Islands  -1.165232   
660672                     Daylight                Highland  -3.852421   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
...                      ...                 ...                     ...   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   
660672                     1                   2                    Snow   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
5       Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
660629  Single carriageway               Rural      Fine + high winds   
660645  Single carriageway               Rural      Fine + high winds   
660672  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
5                                        Car  
...                                      ...  
660629                                   Car  
660645                                   Car  
660672                                   Car  
660677                 Motorcycle over 500cc  
660

In [12]:
uk_accident['Vehicle_Type'].unique()

array(['Car', 'Taxi/Private hire car',
       'Bus or coach (17 or more pass seats)', 'Other vehicle',
       'Van / Goods 3.5 tonnes mgw or under',
       'Motorcycle over 125cc and up to 500cc',
       'Goods 7.5 tonnes mgw and over', 'Motorcycle 125cc and under',
       'Motorcycle over 500cc', 'Motorcycle 50cc and under',
       'Minibus (8 - 16 passenger seats)',
       'Goods over 3.5t. and under 7.5t', 'Pedal cycle',
       'Agricultural vehicle', 'Ridden horse',
       'Data missing or out of range'], dtype=object)

In [13]:
uk_accident['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

In [14]:
uk_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [15]:
uk_accident['Latitude'] = uk_accident['Latitude'].fillna(uk_accident['Latitude'].mean())
uk_accident['Longitude'] = uk_accident['Longitude'].fillna(uk_accident['Longitude'].mean())
uk_accident['Road_Surface_Conditions'] = uk_accident['Road_Surface_Conditions'].fillna(uk_accident['Road_Surface_Conditions'].mode()[0])
uk_accident['Road_Type'] = uk_accident['Road_Type'].fillna(uk_accident['Road_Type'].mode()[0])
uk_accident['Urban_or_Rural_Area'] = uk_accident['Urban_or_Rural_Area'].fillna(uk_accident['Urban_or_Rural_Area'].mode()[0])
uk_accident['Weather_Conditions'] = uk_accident['Weather_Conditions'].fillna(uk_accident['Weather_Conditions'].mode()[0])

In [16]:
uk_accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

In [17]:
uk_accident['Accident Date'] = pd.to_datetime(uk_accident['Accident Date'], dayfirst = True, errors = 'coerce')

In [18]:
uk_accident.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
dtype: object

In [19]:
uk_accident['Year'] = uk_accident['Accident Date'].dt.year
uk_accident['Month'] = uk_accident['Accident Date'].dt.month
uk_accident['Day'] = uk_accident['Accident Date'].dt.day
uk_accident['DayOfWeek'] = uk_accident['Accident Date'].dt.dayofweek

In [20]:
uk_accident.isnull().sum()

Index                           0
Accident_Severity               0
Accident Date              395672
Latitude                        0
Light_Conditions                0
District Area                   0
Longitude                       0
Number_of_Casualties            0
Number_of_Vehicles              0
Road_Surface_Conditions         0
Road_Type                       0
Urban_or_Rural_Area             0
Weather_Conditions              0
Vehicle_Type                    0
Year                       395672
Month                      395672
Day                        395672
DayOfWeek                  395672
dtype: int64

<h1> CHANGE THE DATA TYPES </h1>

In [21]:
uk_accident['Accident_Severity'] = uk_accident['Accident_Severity'].astype('category')
uk_accident['Light_Conditions'] = uk_accident['Light_Conditions'].astype('category')
uk_accident['District Area'] = uk_accident['District Area'].astype('category')
uk_accident['Road_Surface_Conditions'] = uk_accident['Road_Surface_Conditions'].astype('category')
uk_accident['Road_Type'] = uk_accident['Road_Type'].astype('category')
uk_accident['Urban_or_Rural_Area'] = uk_accident['Urban_or_Rural_Area'].astype('category')
uk_accident['Weather_Conditions'] = uk_accident['Weather_Conditions'].astype('category')
uk_accident['Vehicle_Type'] = uk_accident['Vehicle_Type'].astype('category')

In [22]:
uk_accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 18 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   Index                    660679 non-null  object        
 1   Accident_Severity        660679 non-null  category      
 2   Accident Date            265007 non-null  datetime64[ns]
 3   Latitude                 660679 non-null  float64       
 4   Light_Conditions         660679 non-null  category      
 5   District Area            660679 non-null  category      
 6   Longitude                660679 non-null  float64       
 7   Number_of_Casualties     660679 non-null  int64         
 8   Number_of_Vehicles       660679 non-null  int64         
 9   Road_Surface_Conditions  660679 non-null  category      
 10  Road_Type                660679 non-null  category      
 11  Urban_or_Rural_Area      660679 non-null  category      
 12  Weather_Conditio

<h1> QUESTIONS WITH INSIGHT FOR THE UK ROAD ACCIDENT </h1>

<hr>

<h1> 1. WHICH ACCIDENT SEVERITY TYPE APPEARS MOST OFTEN? </h1>

In [23]:
uk_accident['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h1> INSIGHT #1 </h1>

<h3> SLIGHT ACCIDENT APPEAR THE MOST, WHILE FATAL ONES ARE THE LEAST.</h3>

<hr>

<h1> 2. WHICH LIGHT CONDITION IS THE MOST COMMON DURING ACCIDENTS? </h1>

In [24]:
uk_accident['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h1> INSIGHT #2 </h1>

<h3> DAYLIGHT ACCIDENTS ARE THE MOST FREQUENT DUE TO HIGHER TRAFFIC FLOW. </h3>

<hr>

<h1> 3. WHAT IS THE AVERAGE NUMBER OF CASUALTIES PER ACCIDENT? </h1>

In [25]:
uk_accident['Number_of_Casualties'].mean()

np.float64(1.357040257068864)

<h1> INSIGHT #3 </h1>

<h3> EACH ACCIDENT RESULT IN AT LEAST ONE CASUALTY ON AVERAGE.</h3>

<hr>

<h1> 4. WHAT IS THE AVERAGE NUMBER OF CASUALTIES IN DAYLIGHT ACCIDENTS? </h1>

In [26]:
uk_accident['DayOfWeek'] = pd.to_datetime(uk_accident['Accident Date'], errors='coerce').dt.day_name()
uk_accident['DayOfWeek'].value_counts()

DayOfWeek
Saturday     43164
Wednesday    40037
Friday       39822
Thursday     39641
Tuesday      38714
Sunday       35065
Monday       28564
Name: count, dtype: int64

<h1> INSIGHT #4 </h1>

<h3> MOST ACCIDENTS HAPPENED ON SATURDAYS, WHILE MONDAYS HAD THE FEWEST REPORTED ACCIDENT.  </h3>

<hr>

<h1> 5. WHICH VEHICLE TYPE HAS THE HIGHEST AVERAGE CASUALTIES? </h1>

In [27]:
avg_casualties_vehicle = uk_accident.groupby('Vehicle_Type')['Number_of_Casualties'].mean()

In [28]:
avg_casualties_vehicle

Vehicle_Type
Agricultural vehicle                     1.342065
Bus or coach (17 or more pass seats)     1.349216
Car                                      1.358841
Data missing or out of range             1.500000
Goods 7.5 tonnes mgw and over            1.351881
Goods over 3.5t. and under 7.5t          1.362861
Minibus (8 - 16 passenger seats)         1.345648
Motorcycle 125cc and under               1.332635
Motorcycle 50cc and under                1.337235
Motorcycle over 125cc and up to 500cc    1.343521
Motorcycle over 500cc                    1.359434
Other vehicle                            1.340075
Pedal cycle                              1.370558
Ridden horse                             1.250000
Taxi/Private hire car                    1.368663
Van / Goods 3.5 tonnes mgw or under      1.354537
Name: Number_of_Casualties, dtype: float64

<h1> INSIGHT #5  </h1>

<h3> DATA MISSING OR OUT OF RANGE TYPE OF VEHICLE HAS THE HIGHEST AVERAGE CASUALTIES. </h3>

<hr>

<h1> 6. WHICH YEAR RECORDED THE MOST ACCIDENT? </h1>

In [29]:
uk_accident['Year'] = pd.to_datetime(uk_accident['Accident Date'], errors='coerce').dt.year
uk_accident['Year'].value_counts()

Year
2019.0    71867
2020.0    70163
2021.0    66172
2022.0    56805
Name: count, dtype: int64

<h1> INSIGHT #6 </h1>

<h3> 2019 HAS THE MOST RECORDED ACCIDENT WITH A 71,867 RECORDS. </h3>

<hr>

<h1> 7.  WHICH DISTRICT HAS THE HIGHEST AVERAGE CASUALTIES?  </h1>

In [30]:
district_avg_casualties = uk_accident.groupby('District Area')['Number_of_Casualties'].mean()
district_avg_casualties

District Area
Aberdeen City    1.139834
Aberdeenshire    1.310363
Adur             1.319871
Allerdale        1.474291
Alnwick          1.512931
                   ...   
Wychavon         1.427627
Wycombe          1.412377
Wyre             1.378029
Wyre Forest      1.382869
York             1.293094
Name: Number_of_Casualties, Length: 422, dtype: float64

<h1> INSIGHT#7 </h1>


<h3> ALNWICK EXPERIENCE THE MOST SEVERE ACCIDENTS ON AVERAGE. </h3>

<hr>

<h1> 8. WHICH DISTRICT HAS THE HIGHEST AVERAGE VEHICLES INVOLVED?</h1>

In [31]:
district_avg_vehicles = uk_accident.groupby('District Area')['Number_of_Vehicles'].mean()
district_avg_vehicles

District Area
Aberdeen City    1.637188
Aberdeenshire    1.553368
Adur             1.969305
Allerdale        1.782801
Alnwick          1.646552
                   ...   
Wychavon         1.828068
Wycombe          1.866397
Wyre             1.860258
Wyre Forest      1.789474
York             1.856089
Name: Number_of_Vehicles, Length: 422, dtype: float64

<h1> INSIGHT #8 </h1>

<h3> BASED ON THE DATA, ADUR HAVE MULTI-VEHICLE ACCIDENTS. </h3>

<hr>

<h1> 9. IS THERE A RELATIONSHIP BETWEEN NUMBER OF VEHICLES AND NUMBER OF CASUALTIES?   </h1>

In [32]:
vehicles_casualties_corr = uk_accident['Number_of_Vehicles'].corr(uk_accident['Number_of_Casualties'])
np.round(vehicles_casualties_corr,2)

np.float64(0.23)

<h1> INSIGHT #9 </h1>

<h3> IT'S POSITIVE VALUE MEANS MORE VEHICLES SLIGHTLY INCREASE CASUALTIES </h3>

<hr>

<h1> 10. WHAT IS THE MAXIMUM NUMBER OF VEHICLES IN THE ENTIRE DATASET? </h1>

In [33]:
max_vehicles_overall = uk_accident['Number_of_Vehicles'].max()
max_vehicles_overall

np.int64(32)

<h1> INSIGHT #10 </h1> 

<h3> THE MAXIMUM NUMBER OF VEHICLES RECORDED IN A SINGLE ACCIDENT IS 32. </h3>

<hr>

<h1> 11. WHICH ROAD SURFACE CONDITION HAS THE HIGHEST AVERAGE CASUALTIES? </h1>

In [34]:
road_avg_casualties = uk_accident.groupby('Road_Surface_Conditions')['Number_of_Casualties'].mean()
road_avg_casualties

Road_Surface_Conditions
Dry                     1.334545
Flood over 3cm. deep    1.495575
Frost or ice            1.348815
Snow                    1.359593
Wet or damp             1.411064
Name: Number_of_Casualties, dtype: float64

<h1> INSIGHT#11 </h1>

<h3> THE FLOODED ROADS HAVE THE HIGHEST AVERAGE CASUALTIES COMPARED TO OTHER SURFACE CONDITIONS. </h3>

<hr>

<h1> 12. WHICH MONTH RECORDED THE MOST ACCIDENTS? </h1>

In [35]:
uk_accident['Month'] = uk_accident['Accident Date'].dt.month
uk_accident['Month'].value_counts()

Month
11.0    24240
12.0    24156
10.0    23962
7.0     22939
9.0     22558
2.0     22264
6.0     22196
3.0     21824
5.0     21723
8.0     21106
4.0     19787
1.0     18252
Name: count, dtype: int64

<h1> INSIGHT#12 </h1>

<h3> NOVEMBER SHOWS WHEN ACCIDENTS ARE MORE FREQUENT, POSSIBLY DUE TO SEASONAL EFFECTS. </h3>

<hr>

<h1> 13. WHICH DISTRICT HAS THE MOST FATAL ACCIDENTS? </h1>

In [37]:
uk_accident[uk_accident['Accident_Severity'] == 'Fatal']['District Area'].value_counts()

District Area
Birmingham                  105
Leeds                        93
Highland                     88
East Riding of Yorkshire     85
Bradford                     71
                           ... 
Orkney Islands                2
Havant                        2
Stevenage                     1
Clackmannanshire              1
Harlow                        1
Name: count, Length: 422, dtype: int64

<h1> INSIGHT#13 </h1> 

<h3> BIRMINGHAM HAS THE HIGHEST NUMBER OF FATAL ACCIDENTS SHOWING IT IS THE MOST CRITICAL AREA FOR ROAD SAFETY COMPARED TO OTHER DISTRICTS. </h3>

<hr>

<h1> 14. IS THERE A RELATIONSHIP BETWEEN VEHICLES AND MONTH? </h1>

In [38]:
month_vehicle_corr = uk_accident['Month'].corr(uk_accident['Number_of_Vehicles'])
month_vehicle_corr

np.float64(0.008613400083189825)

<h1> INSIGHT#14 </h1>

<h3> THE RESULT IS CLOSE TO ZERO, SO VEHICLES AND CASUALTIES HAVE ALMOST NO RELATIONSHIP. </h3>

<hr>

<h1> 15. WHAT IS THE AVERAGE NUMBER OF CASUALTIES IN URBAN VS RURAL AREAS? </h1>

In [40]:
uk_accident.groupby('Urban_or_Rural_Area')['Number_of_Casualties'].mean()

Urban_or_Rural_Area
Rural          1.479204
Unallocated    1.181818
Urban          1.287807
Name: Number_of_Casualties, dtype: float64

<h1> INSIGHT #15 </h1>

<h3> RURAL AREAS HAVE THE HIGHEST AVERAGE CASUALTIES MEANS ACCIDENTS IN RURAL LOCATIONS TEND TO BE MORE SEVERE THAN URBAN OR UNALLOCATED AREAS. </h3>

<hr>

<h1> 16. WHAT IS THE AVERAGE NUMBER OF CASUALTIES IN URBAN DAYLIGHT ACCIDENTS? </h1>

In [42]:
uk_accident[(uk_accident['Urban_or_Rural_Area']=='Urban') & (uk_accident['Light_Conditions']=='Daylight')]['Number_of_Casualties'].mean()

np.float64(1.2688250554889653)

<h1> INSIGHT #16  </h1>

<h3> URBAN DAYLIGHT ACCIDENTS USUALLY HAVE ABOUT 1 TO 2 CASUALTIES. </h3>

<hr>

<h1> 17. WHICH DISTRICT HAS THE FEWEST RECORDED ACCIDENTS? </h1>

In [43]:
uk_accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h1> INSIGHT 17 </h1>

<h3> CLACKMANNANSHIRE HAVE THE FEWEST ACCIDENTS DUE TO SMALLER POPULATION AND ROAD SIZE.</h3>

<hr>

<h1> 18. WHICH WEATHER CONDITION IS LEAST COMMON DURING ACCIDENTS?  </h1>

In [44]:
uk_accident['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       535013
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h1> INSIGHTS #18 </h1>

<h3> SNOWING WITH HIGH WINDS HAS THE FEWEST RECORDED ACCIDENTS 885, SHOWING THAT THIS WEATHER CONDITION IS RARE IN THE DATASET. </h3>

<hr>

<h1> 19. WHAT IS THE CORRELATION BETWEEN ACCIDENT SEVERITY AND NUMBER OF VEHICLES? </h1>

In [45]:
uk_accident['Accident_Severity'].cat.codes.corr(uk_accident['Number_of_Vehicles'])

np.float64(0.07532400982810716)

<h1> INSIGHT #19 </h1> 

<h3> WITH A LOW CORRELATION 0.07, VEHICLE COUNT DOES NOT STRONGLY IMPACT HOW SEVERE AN ACCIDENT IS. </h3>

<hr>

<h1> 20. WHICH DISTRICT RECORDS THE HIGHEST NUMBER OF SLIGHT ACCIDENTS? </h1>

In [46]:
uk_accident.groupby('District Area')['Accident_Severity'].value_counts()

District Area  Accident_Severity
Aberdeen City  Slight               1072
               Serious               239
               Fatal                  12
Aberdeenshire  Slight               1401
               Serious               463
                                    ... 
Wyre Forest    Serious               132
               Fatal                  22
York           Slight               1621
               Serious               255
               Fatal                  21
Name: count, Length: 1266, dtype: int64

<h1> INSIGHT #20 </h1>

<h3> YORK RECORDED THE HIGHEST SLIGHT ACCIDENTS 1621, SHOWING THAT MINOR COLLISIONS ARE MORE FREQUENT IN THIS AREA COMPARED TO OTHERS. </h3>

<hr>

<h1> 21. WHAT IS THE MINIMUM NUMBER OF CASUALTIES IN ACCIDENTS ON DRY ROAD?</h1>

In [52]:
uk_accident[uk_accident['Road_Surface_Conditions'] == 'Dry']['Number_of_Casualties'].min()

np.int64(1)

<h1> INSIGHT #21 </h1> 

<h3> THE MINIMUM NUMBER OF CASUALTIES RECORDED ON DRY ROADS IS 1.</h3>

<hr>

<h1> 22. WHAT IS THE COUNT OF EACH VEHICLE IN RURAL AREAS? </h1>

In [53]:
uk_accident[uk_accident['Urban_or_Rural_Area'] == 'Rural']['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      181922
Van / Goods 3.5 tonnes mgw or under       12156
Bus or coach (17 or more pass seats)       9025
Motorcycle over 500cc                      8957
Goods 7.5 tonnes mgw and over              6156
Motorcycle 125cc and under                 5023
Taxi/Private hire car                      4677
Motorcycle 50cc and under                  2710
Motorcycle over 125cc and up to 500cc      2674
Goods over 3.5t. and under 7.5t            2232
Other vehicle                              1994
Minibus (8 - 16 passenger seats)            718
Agricultural vehicle                        675
Pedal cycle                                  70
Ridden horse                                  1
Data missing or out of range                  0
Name: count, dtype: int64

<h1> INSIGHT #22 </h1> 

<h3> CARS MAKE UP THE MAJORITY OF RURAL ACCIDENTS, WHILE VEHICLES LIKE HORSES OR BICYCLES ARE RARELY INVOLVED. </h3>

<hr>

<h1> 23. IN THE WYRE DISTRICT, WHICH ROAD TYPE HAS THE LEAST ROAD ACCIDENT?  </h1>

In [55]:
uk_accident[uk_accident["District Area"]=="Wyre"]["Road_Type"].value_counts()

Road_Type
Single carriageway    1004
Roundabout              94
Dual carriageway        93
One way street          45
Slip road                2
Name: count, dtype: int64

<h1> INSIGHT #23 </h1> 

<h3> SLIP ROADS IN WYRE HAVE THE LEAST ACCIDENTS 2, SHOWING THEY ARE RARELY INVOLVED IN CRASHES. </h3>

<hr>

<h1> 24. WHAT IS THE LOWEST CASUALTIES BY ROAD TYPE? </h1>

In [59]:
uk_accident.groupby('Road_Type')['Number_of_Casualties'].max()

Road_Type
Dual carriageway      47
One way street        15
Roundabout            19
Single carriageway    62
Slip road             68
Name: Number_of_Casualties, dtype: int64

<h1> INSIGHT #24 </h1>

<h3> ONE-WAY STREETS HAVE THE LOWEST MAXIMUM CASUALTIES 15, SHOWING THEY ARE SAFER COMPARED TO OTHER ROAD TYPES.  </h3>

<hr>

<h1> 25. WHAT IS THE MEDIAN NUMBERS OF CASUALTIES IN RURAL AREAS? </h1>

In [61]:
uk_accident[uk_accident['Urban_or_Rural_Area'] == 'Rural']['Number_of_Casualties'].median()

np.float64(1.0)

<h1> INSIGHT #25 </h1>

<h3> THE MEDIAN NUMBER OF CASUALTIES IN RURAL AREAS IS 1. </h3>

<hr> 

<h1> 26. WHAT DATE HAVE THE FEWEST AMOUNT OF ACCIDENTS IN A SINGLE DAY? </h1>

In [62]:
uk_accident['Accident Date'].value_counts()

Accident Date
2021-02-11    685
2021-10-06    680
2019-12-06    678
2019-02-01    676
2021-12-04    667
             ... 
2022-01-09    236
2022-12-02    236
2022-04-05    231
2022-02-07    213
2022-01-10    123
Name: count, Length: 576, dtype: int64

<h1> INSIGHT #26  </h1> 

<h3> 2022-01-10 HAD THE FEWEST ACCIDENTS 123, SHOWING IT WAS AN UNUSUALLY SAFE DAY COMPARED TO OTHER DATES. </h3>

<hr>